alf/utils/common.py

```diff
@@ -370,16 +370,21 @@ def process(line):
 
 
 def summarize_gin_config():
-    """Write the operative gin config to Tensorboard summary.
+    """Write the operative and inoperative gin config to Tensorboard summary.
     
     The operative configuration consists of all parameter values used by
     configurable functions that are actually called during execution of the
-    current program. See `gin.operative_config_str()` for more detail on how
-    the operative config is generated.
+    current program, and inoperative configuration consists of all parameter
+    configured but not used by configurable functions. See `gin.operative_config_str()`
+    and `gin_utils.inoperative_config_str` for more detail on how the config is generated.
     """
-    config_str = gin.operative_config_str()
-    md_config_str = _markdownify_operative_config_str(config_str)
+    operative_config_str = gin.operative_config_str()
+    md_config_str = _markdownify_gin_config_str(operative_config_str)
     tf.summary.text('gin/operative_config', md_config_str)
+    inoperative_config_str = gin_utils.inoperative_config_str()
+    if inoperative_config_str:
+        md_config_str = _markdownify_gin_config_str(inoperative_config_str)
+        tf.summary.text('gin/inoperative_config', md_config_str)
```

Comment:
Maybe write a more descriptive text for `inoperative_config`? For example, "A config string capturing all parameter values configured but not used by the current program (override by explicit call)." So this text is well explained and a person won't get confused even without looking at the code.

Reply:
`in_utils.inoperative_config_str.__doc__` is not appropriate here and explicit str description is too long  

We use gin config with all experiments , and should be familiar with parameters configured at least,   so it should not be a problem ...

Reply:
My point is that people might not be familiar with the concept of "inoperative config" without looking into gin's code. At least we can say several words like "config overridden by program" to make it clear.

Reply:
added at the head  of text summary